src/app/modules/form/form.errors.ts

```diff
@@ -60,3 +62,26 @@ export class LogicNotFoundError extends ApplicationError {
     super(message)
   }
 }
+
+/**
+ * Error to be returned when the form's auth type does not match what is required
+ */
+export class AuthTypeMismatchError extends ApplicationError {
+  constructor(attemptedAuthType: AuthType, formAuthType?: AuthType) {
+    super(
+      `Attempted authentication type ${attemptedAuthType} did not match form auth type ${formAuthType}`,
+    )
+  }
+}
+
+/**
+ * Error to be returned when the form is lacking an eServiceId
```

Comment:
nit: to be more precise, you can add that the form has SP, CP or MyInfo enabled but is missing an e-service ID